dbcon/joblist/jlf_tuplejoblist.cpp

```diff
@@ -2374,10 +2903,12 @@ void matchEdgesInRowGroup(const JobInfo& jobInfo, const RowGroup& rg, JoinEdges&
   }
 
   std::vector<pair<uint32_t, uint32_t>> takenEdges;
-  for (const auto& edge : edgesToTransform)
+  for (const auto& joinEdgeWithWeight : edgesToRestore)
```

Comment:
You might be interested in structured binding here to produce edge and weight as a separate variables available in the loop body.

Reply:
Thanks, fixed!